TeamsSampleHub/SampleHub/Manifest/manifest.json

```diff
@@ -28,17 +28,26 @@
     {
       "entityId": "Hub",
       "name": "Hub",
-      "contentUrl": "{url}",
+      "contentUrl": "https://{{appdomain}}.azurewebsites.net",
       "scopes": [ "personal" ]
     },
     {
       "entityId": "Learning",
       "name": "Learning",
-      "contentUrl": "{url}",
+      "contentUrl": "https://{{appdomain}}.azurewebsites.net/learning",
       "scopes": [ "personal" ]
     }
   ],
   "permissions": ["identity", "messageTeamMembers"],
-  "validDomains": [""]
-  
+  "validDomains": [ "*.ngrok.io",    
+  "*.powerapps.com",    
+  "youtu.be",    
+  "*.youtube.com",    
+  "{{appDomain}}.azurewebsites.net",    
+  "token.botframework.com",    
+  "{{spDomain}}.sharepoint.com"],
```

Comment:
I'm wondering if you can't also use the `https://{teamSiteDomain}` placeholder here instead of `{{spDomain}}.sharepoint.com`. It should work in theory, but you may have to try it out to verify that it does in fact add the subdomain.sharepoint.com domain to the validDomains list by seeing the SP site is allowed to be iframed.